specification/azsadmin/resource-manager/compute/Microsoft.Compute.Admin/preview/2018-07-30-preview/Disks.json

```diff
@@ -0,0 +1,319 @@
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "Disks.json#/definitions/DiskProperties",
+          "description": "Disk properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../2015-12-01-preview/Compute.json#/definitions/Resource"
+        }
+      ]
+    },
+    "DiskList": {
+      "type": "object",
+      "description": "List of disks.",
+      "properties": {
+        "value": {
+          "description": "List of disks.",
+          "type": "array",
+          "items": {
+            "$ref": "Disks.json#/definitions/Disk"
+          }
+        },
+        "nextLink": {
+          "description": "URI to the next page.",
+          "type": "string"
+        }
+      }
+    },
+    "DiskState": {
+      "description": "Disk State.",
+      "type": "string",
+      "enum": [
+        "Undefined",
+        "Unattached",
+        "Attached",
+        "Reserved",
+        "ActiveSAS",
+        "Unknown",
+        "All",
+        "Recommended",
+        "OfflineMigration",
+        "OnlineMigration"
+      ],
+      "x-ms-enum": {
+        "name": "DiskState",
+        "modelAsString": true
+      }
+    },
+    "DiskResourceType": {
+      "description": "Disk resource type.",
+      "type": "string",
+      "enum": [
+        "Undefined",
+        "Disk",
+        "Snapshot",
+        "RestorePoint",
+        "ManagedBlob"
+      ],
+      "x-ms-enum": {
+        "name": "DiskResourceType",
+        "modelAsString": true
+      }
+    },
+    "AccountType": {
+      "description": "Disk Sku.",
+      "type": "string",
+      "enum": [
+        "Standard_LRS",
+        "Standard_ZRS",
+        "Standard_GRS",
+        "Standard_RAGRS",
+        "Premium_LRS",
+        "StandardSSD_LRS",
+        "UltraSSD_LRS"
+      ],
+      "x-ms-enum": {
+        "name": "AccountType",
+        "modelAsString": true
+      }
+    }
+  },
+  "parameters": {
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "description": "Client API Version.",
+      "required": true,
+      "type": "string",
+      "default": "2018-07-30-preview"
+    },
+    "TenantSubscriptionIdParameter": {
+      "description": "Tenant Subscription Id which the resource belongs to.",
+      "name": "tenantSubscriptionId",
+      "type": "string",
+      "in": "query",
+      "required": false,
+      "x-ms-parameter-location": "method"
+    },
+    "DiskStateParameter": {
+      "description": "The parameters of disk state.",
+      "name": "status",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "SourceSharePathParameter": {
+      "description": "The source share which the resource belongs to.",
+      "name": "sharePath",
+      "in": "query",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "DiskCountParameter": {
```

Comment:
Is there a client that requested these two parameters (Start and Count)? The standard practice is to use continuation tokens when going through the large result set. Try to avoid introducing inconsistent patterns.

Reply:
I will open a work item to check. Currently the service side doesn't support this.